congame-doc/introduction.scrbl

```diff
@@ -1,37 +1,530 @@
+functionality as well as explanatory text.
+
+Working from the inside out:
+
+@itemlist[#:style 'ordered
+
+@item{The @racket[(input-text)] and @racket[(input-number)] expressions (at the end of the fifth and
+seventh lines in the code above) generate input boxes where the participant can enter data. There
+are other expressions that can be used to insert the various other form elements like checkboxes and
+dropdown lists.}
+          
+@item{The @racket[(set! first-name (input-text))] expression tells Conscript, “when this form is
+submitted, set the value in the @racket[_first-name] variable to the whatever the user has entered
+in @racket[(input-text)]. }
+
+@item{The @racket[submit-button] expression inserts the button the user can click to submit all the
+information they have entered into the form elements.}
+
+@item{All the interactive form elements are wrapped in a @racket[form] expression to keep them
+grouped together.}
+
+]
+
+Next, write the final step by adding these lines to the end of your file:
+
+@codeblock[#:keep-lang-line? #f]|{
+#lang conscript/local
+
+(defstep (thank-you)
+  @md{
+    # Good job, @first-name
+
+    Thank you for participating in our survey despite
+    being @number->string[age] years old.
+  })
+}|
+
+We’re doing something else interesting here: we’re inserting our variables directly into the text in
+the @racket[md] expression! We’ve told Conscript to record @racket[_first-name] and @racket[_age] in
+the previous step of the study, but in addition to storing those values for later analysis, we can
+make use of those values while the user is still participating. There are many ways we might want to
+do this; in this case, we’re simply displaying the values back to the user.
+
+@margin-note{Note that since we used @racket[input-number] to assign a value to @racket[_age], we
+must use @racket[number->string] to convert that numeric value to text if we want to display it
+within @racket[md].}
+
+Finally, we’ll define the study as a whole by tying all the steps together in a transition graph:
+
+@codeblock[#:keep-lang-line? #f]|{
+#lang conscript/local
+
+(defstudy simple-survey
+  [description --> age-name-survey --> thank-you]
+  [thank-you --> thank-you])
+
+(provide simple-survey)
+
+}|
+
+We’ve now told Conscript that our study is named @racket[simple-survey], and that it consists of
+three steps @racket[description], @racket[age-name-survey] and @racket[thank-you] in that order.
+Because every step must have a transition, even the last one, we add @racket[[thank-you -->
+thank-you]] to tell Conscript that that step simply transitions to itself.
+
+The last line is not technically required at @emph{this} point, but it will be needed later when we
+upload our study to a Congame server. Without it, the server will not be able to access the study
+bound to the @racket[simple-study] identifier. 
+
+@margin-note{It’s a good practice to add @racket[(provide _studyname)] following every
```

Comment:
I usually do `provide` at the top or before the definitions. And I would remove the `(provide (all-defined-out))`, I prefer being explicit about this.